### PR TITLE
Editor: Introduce QueryPublicizeConnections to replace EditorSharingContainer

### PR DESCRIPTION
**Pull request by aduth**
This pull request seeks to introduce a new `<QueryPublicizeConnections />` [query component](https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md#query-components) to be used in managing requests for Publicize connections. See the [component README](https://github.com/Automattic/wp-calypso/blob/add/query-publicize-connections/client/components/data/query-publicize-connections/README.md) for more information and usage. The query component is used in place of the previously implemented `<EditorSharingContainer />` (#756), as this is now our preferred method for introducing data to a component hierarchy.

__Implementation notes:__

Future pull requests will...
- Further improve the consumption of data from the Redux tree. Specifically, the accordion subtitle should be split to its own component, and both this new component and `<EditorSharingPublicizeOptions />` should separately bind to the Redux state and actions with their specific data needs.
- Complete the transition away from "fetching" terminology to "requesting", hence the `import ... as` syntax included in the changes here.

__Testing instructions:__

Repeat testing instructions from #756.

### CODE REVIEW

File: client/components/data/query-publicize-connections/index.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import { isFetchingConnections as isRequestingConnections } from 'state/sharing/publicize/selectors';
+import { fetchConnections as requestConnections } from 'state/sharing/publicize/actions';
```
Comment (timmyc) [inline client/components/data/query-publicize-connections/index.jsx]: Why the naming swaps here, just curious?

File: client/post-editor/editor-sharing/accordion.jsx [hunk 2]
```diff
@@ -1,22 +1,29 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	Gridicon = require( 'components/gridicon' ),
-	classNames = require( 'classnames' );
+import React from 'react';
+import Gridicon from 'components/gridicon';
```
Comment (timmyc) [inline client/post-editor/editor-sharing/accordion.jsx]: Should `Gridicon` be under internal dependencies?

File: client/post-editor/editor-sharing/accordion.jsx [hunk 3]
```diff
@@ -126,3 +136,20 @@ module.exports = React.createClass( {
 		);
 	}
 } );
+
```
Comment (timmyc) [inline client/post-editor/editor-sharing/accordion.jsx]: There is still quite a bit of non-es6 syntax in this file, wasn't sure if you wanted to cleanup all the `var` usage in here while updating or grab it in a future PR.

File: client/components/data/query-publicize-connections/index.jsx (hunk 1 above)
Comment (timmyc) [inline client/components/data/query-publicize-connections/index.jsx]: nm

> Complete the transition away from "fetching" terminology to "requesting", hence the import ... as syntax included in the changes here.

**Comment by timmyc** [comment]
Tests out well - pretty nice to toggle between two sites that already have the connection data in the global state tree - very snappy re-renders in the editor sidebar!

Couple of minor comments above, feel free to address or save for later and :ship: now.

File: client/post-editor/editor-sharing/accordion.jsx (hunk 2 above)
Comment (aduth) [inline client/post-editor/editor-sharing/accordion.jsx]: >Should `Gridicon` be under internal dependencies?

Yep, definitely should. Moved in rebased ce609d10c3defb00404acd49a3e0b59aa75facac.

File: client/post-editor/editor-sharing/accordion.jsx (hunk 3 above)
Comment (aduth) [inline client/post-editor/editor-sharing/accordion.jsx]: >There is still quite a bit of non-es6 syntax in this file

Similar to remarks at https://github.com/Automattic/wp-calypso/pull/3197#discussion_r52539734, I'll probably leave the remaining legacy for future pull requests.

**Pull request merged**